### PR TITLE
[DEMOS] chore: updated react-native-supabase-todolist demo to use expo environment variables

### PR DESCRIPTION
**Pull request by Chriztiaan**
# Description
Updated `react-native-supabase-todolist` to support EAS.

Instead of configuring the variables in the `AppConfig.ts` file directly, configuration is done via a `.env` or `.env.local` file.

Proof of EAS build working after configuring the `.env` file with my project credentials:
image

# Known issue
The EAS Android build currently has an issue with login with the Supabase client. I have confirmed that the .env configs are loaded and the username and password are non-empty strings at the point where login is done with the SB client.

Screenshot 2024-03-18 at 13 03 09

# Outstanding work
- [x]  .changeset

### CODE REVIEW

File: demos/react-native-supabase-group-chat/.env [hunk 1]
```diff
@@ -1,5 +1,6 @@
-# Replace the credentials below with you Supabase, PowerSync and Expo project details.
+# Copy this template: `cp .env.template .env`
```
Comment (stevensJourney) [inline demos/react-native-supabase-group-chat/.env:1]: Should this not say `cp .env .env.local`?

File: demos/react-native-supabase-todolist/.env [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Edit and enter your Supabase and PowerSync project details.
+EXPO_PUBLIC_SUPABASE_URL=https://foo.supabase.co
+EXPO_PUBLIC_SUPABASE_ANON_KEY=foo
+EXPO_PUBLIC_SUPABASE_BUCKET= // Optional. Only required when syncing attachments and using Supabase Storage. See packages/powersync-attachments.
```
Comment (stevensJourney) [inline demos/react-native-supabase-todolist/.env:4]: The commend should begin with `#`

File: demos/react-native-supabase-group-chat/.env (hunk 1 above)
Comment (Chriztiaan) [reply demos/react-native-supabase-group-chat/.env:1]: Apparently I never saved the file.

**Review by stevensJourney — DISMISSED** [review]
It's unfortunate that there is a runtime issue with the app when running the EAS build. I could not reproduce this issue locally. Having the environment variables pulling through to the build correctly is probably okay for this project/work.

File: demos/react-native-supabase-todolist/README.md [hunk 3]
```diff
@@ -36,14 +36,16 @@ bucket_definitions:
 ```
 
 ## Configure The App
+Replace the necessary credentials in the [.env](./.env) file.
+Generally, the `.env` file is used for storing common environment variables shared across all instances of the application, while `.env.local` is for overriding or providing environment-specific configurations, particularly for local development.
+As `.env.local` is normally not checked into source control (this project has a git-ignore rule), you can copy `.env`, name it `.env.local`, and then configure as needed.
 
-Copy the `AppConfig.template.ts` to a usable file
+### EAS Build configuration
+Take note that you will need an Expo account if you want to use EAS for your builds, the Expo project ID should then also be configured in the environment file.
```
Comment (benitav) [inline demos/react-native-supabase-todolist/README.md:44]:
```suggestion
Take note that you will need an [Expo](https://expo.dev/) account if you want to use EAS for your builds, the Expo project ID should then also be configured in the environment file.
```

File: demos/react-native-supabase-todolist/README.md [hunk 4]
```diff
@@ -64,3 +66,12 @@ Run on Android
 ```sh
 pnpm android
 ```
+
+
+## Here are some helpful links:
+
+- [PowerSync Website](https://www.powersync.com/)
+- [PowerSync Docs](https://docs.powersync.com/)
+- [PowerSync React Native Client SDK](https://github.com/journeyapps/powersync-js/tree/main/packages/powersync-sdk-react-native)
```
Comment (benitav) [inline demos/react-native-supabase-todolist/README.md:75]:
```suggestion
- [PowerSync React Native Client SDK Reference](https://docs.powersync.com/client-sdk-references/react-native-and-expo)
```

Comment (benitav) [reply demos/react-native-supabase-todolist/README.md:75]: My thought is this is a higher-level entry to everything a developer might want to know about the SDK - including links to the NPM page, API reference, etc.

File: demos/react-native-supabase-todolist/.env [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Replace the credentials below with you Supabase, PowerSync and Expo project details.
```
Comment (benitav) [inline demos/react-native-supabase-todolist/.env:1]:
```suggestion
# Replace the credentials below with your Supabase, PowerSync and Expo project details.
```

File: demos/react-native-supabase-group-chat/.env [hunk 6]
```diff
@@ -2,4 +2,4 @@
 EXPO_PUBLIC_SUPABASE_URL=https://foo.supabase.co
```
Comment (benitav) [inline demos/react-native-supabase-group-chat/.env:2]: While we're at it we can fix the typo in the line above: you -> your

**Review by stevensJourney — APPROVED** [review]

**Review by benitav — APPROVED** [review]

**Pull request merged**